control-plane-operator/hostedclusterconfigoperator/controllers/machine/machine.go

```diff
@@ -136,29 +158,26 @@ func (r *reconciler) reconcileKubevirtDefaultIngressEndpointsByIPFamily(ctx cont
 	}
 	result, err := r.CreateOrUpdate(ctx, r.kubevirtInfraClient, endpointSlice, func() error {
 		if len(endpointSlice.OwnerReferences) == 0 {
-			// Machine infra ref is the KubevirtMachine wich has the same name
-			// as the kubevirt VirtualMachine CRD, but the namespace ca
-			// can be different if kubevirt infra cluster is external
-			vmKey := client.ObjectKey{
-				Namespace: cpService.Namespace,
-				Name:      machine.Spec.InfrastructureRef.Name,
-			}
-			vm := kubevirtv1.VirtualMachine{}
-			if err := r.kubevirtInfraClient.Get(ctx, vmKey, &vm); err != nil {
-				return err
-			}
-			ownerRef := config.OwnerRefFrom(&vm)
+			ownerRef := config.OwnerRefFrom(cpService)
```

Comment:
This is tricky. We have two scenarios now.

1.passthrough service. In this scenario we know the service will remain throughout the lifetime of the cluster
2. kccm services. In this scenario, the services come and goe throughout the lifetime of the cluster.

Managing the endpoint slices for item 1 is simple because the service itself never goes away which allows us to track ownership using a single dimension (the VM)

Managing endpoint slices for item 2 is complex because we have to track ownership across both the Service and the VM. If the service goes away, all the endpoint slices need to go away, and if the VM goes away, that specific endpoint slice needs to go away.

Here's what I propose.

- Keep the hypershift logic as it is, with endpointslices having a 1:1 owership mapping to VMs.
- Add a endpoint slice garbage collection to kccm. 

The garbage collection would work like this in kccm
1. set finalizer on selectorless services during creation
2. on deletion of service perform the following
    a. mark service for deletion
    b. list and delete all endpoint slices associated with the service
    c. remove finalizer off of service.

Reply:
But what is the scenario that we don't cover having service as the endpoints owner ? 

If machines are gone the Reconcile cycle here will also remove the endpointslices.